File: 0485-max-consecutive-ones/0485-max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let n=nums.length;
    let cnt=0;
    let ans=0;
    for(let i=0;i<n;i++){
        if(nums[i]===0){
            cnt=0;
        }
        else{
            cnt++;
        }
        ans=Math.max(ans,cnt);
    }
    return ans;
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        //max consecutive ones
        int n = nums.size();
        int cnt = 0;
        int ans = 0;
        for(int i=0;i<n;i++){
           if(nums[i]==1){
               cnt++;
           }
            if(nums[i]==0){
                cnt=0;
            }
            ans = max(cnt,ans);
        }
        return ans;
    }
};